tests/foreman/ui/test_computeresource_azurerm.py

```diff
@@ -0,0 +1,250 @@
+    module_org,
+):
+    """ Create hostgroup """
+
+    return entities.HostGroup(
+        architecture=module_architecture,
+        compute_resource=module_azurerm_cr,
+        compute_profile=COMPUTE_PROFILE_SMALL,
+        domain=module_domain,
+        location=[module_loc],
+        environment=module_puppet_environment,
+        puppet_proxy=module_smart_proxy,
+        puppet_ca_proxy=module_smart_proxy,
+        content_source=module_smart_proxy,
+        operatingsystem=module_os,
+        organization=[module_org],
+    ).create()
+
+
+@tier4
+def test_positive_end_to_end_azurerm_ft_host_provision(
+    session,
+    azurermclient,
+    module_azurerm_finishimg,
+    module_azurerm_cr,
+    module_domain,
+    module_org,
+    module_loc,
+    module_azure_hg,
+):
+
+    """Provision Host with hostgroup and Compute-profile using
+    finish template on AzureRm compute resource
+
+    :id: d64d249d-70a2-4329-bff4-3b50b8596c44
+
+    :expectedresults:
+            1. Host is provisioned.
+            2. Host is deleted Successfully.
+
+    :CaseLevel: System
+    """
+
+    hostname = gen_string('alpha')
+    fqdn = '{}.{}'.format(hostname, module_domain.name).lower()
+
+    with session:
+        session.organization.select(org_name=module_org.name)
+        session.location.select(loc_name=module_loc.name)
+        # Provision Host
+        try:
+            skip_yum_update_during_provisioning(template='Kickstart default finish')
+            session.host.create(
+                {
+                    'host.name': hostname,
+                    'host.hostgroup': module_azure_hg.name,
+                    'provider_content.operating_system.root_password': gen_string('alpha'),
+                    'provider_content.operating_system.image': module_azurerm_finishimg.name,
+                }
+            )
+
+            host_info = session.host.get_details(fqdn)
+            assert session.host.search(fqdn)[0]['Name'] == fqdn
+            assert host_info['properties']['properties_table']['Build'] == 'Installed'
+            assert (
+                host_info['properties']['properties_table']['Host group'] == module_azure_hg.name
+            )
+
+            # AzureRm Cloud assertion
+            azurecloud_vm = azurermclient.get_vm(name=hostname.lower())
+            assert azurecloud_vm
+            assert azurecloud_vm.is_running
+            assert azurecloud_vm.name == hostname.lower()
+            assert azurecloud_vm.ip == host_info['properties']['properties_table']['IP Address']
+            assert azurecloud_vm.type == AZURERM_VM_SIZE_DEFAULT
+
+            # Host Delete
+            message = session.host.delete(fqdn)
+            assert (
+                'Are you sure you want to delete host {}? This will delete the VM and '
+                'its disks, and is irreversible. This behavior can be changed via '
+                'global setting "Destroy associated VM on host delete".'.format(fqdn)
+            ) == message
```

Comment:
Do we want this assertion here?
Seems we have that already covered in other tests. And I wouldn't fail this test if exact wording ever changes - seems a little too aggressive for my taste.

Reply:
removed